### PR TITLE
Capacity report

### PR DESCRIPTION
**Pull request by lenhattan86**
## CHANGES

1. pull capacity report via /offers endpoint.
2. calculate how many tasks (with resource and constraints) can be fit in the cluster.

examples of using the above 2 features are in https://github.com/aurora-scheduler/australis/pull/33

## TEST DONE
* unit test
* e2e test
* integration test

### CODE REVIEW

File: offer.go [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+	// validate resource request
+	if len(request) == 0 {
+		return -1, fmt.Errorf("Resource request %v must not be empty", request)
+	}
+
+	isValid := false
+	for _, resVal := range request {
+		if resVal > 0 {
+			isValid = true
+		}
```
Comment (lawwong1) [inline offer.go:232]: break;

File: offer.go [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+	for _, o := range offers {
+		if maintHostSet[o.Hostname] {
+			continue
+		}
+
+		group := "non-dedicated"
+		for _, a := range o.Attributes {
+			if a.Name == "dedicated" {
+				group = a.Text.Value
+			}
```
Comment (lawwong1) [inline offer.go:179]: Add break after line 178;

File: offer.go [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+				matched = true
+			} else {
+				return false
+			}
+		}
+	}
+
+	return matched
+}
+
```
Comment (lawwong1) [inline offer.go:384]: Please add a brief description of what and how the function achieve its work.

File: offer.go [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+			if matched {
+				break
+			}
+		}
+	}
+
+	return matched
+}
+
```
Comment (lawwong1) [inline offer.go:407]: Please add a brief description of what and how the function achieve its work.

File: offer.go (hunk 3 above)
Comment (lenhattan86) [reply offer.go:384]: added will push.

File: offer.go (hunk 1 above)
Comment (lenhattan86) [reply offer.go:232]: good point. will push.

File: offer.go (hunk 2 above)
Comment (lenhattan86) [reply offer.go:179]: added. will push.

File: offer.go (hunk 4 above)
Comment (lenhattan86) [reply offer.go:407]: added. will push.

**Comment by lenhattan86** [comment]
@lawwong1  can you take another pass?

**Review by lawwong1 — APPROVED** [review]
Looks good

**Pull request merged**